src/loaders/abstract-loader.coffee

```diff
@@ -276,23 +277,26 @@ class AbstractLoader
     promises = []
 
     for association in @additionalIncludes
-      loadOptions =
-        cache: @loadOptions.cache
-        headers: @loadOptions.headers
-        only: association.ids
-        params:
-          apply_default_filters: false
-
-      if association.collection
-        promises.push association.collection.fetch(loadOptions)
-      else
-        collectionName = @_getModel().associationDetails(association.name).collectionName
-        if association.loadOptions
-          loadOptions = _.extend(loadOptions, association.loadOptions)
+      batches = Utils.chunk(association.ids, @associationIdLimit)
+
+      batches.forEach (batchedIds) =>
+        loadOptions =
+          cache: @loadOptions.cache
+          headers: @loadOptions.headers
+          only: batchedIds
+          params:
+            apply_default_filters: false
+
+        if association.collection
+          promises.push association.collection.fetch(loadOptions)
         else
-          loadOptions.include = association.include
+          collectionName = @_getModel().associationDetails(association.name).collectionName
+          if association.loadOptions
+            loadOptions = _.extend(loadOptions, association.loadOptions)
+          else
+            loadOptions.include = association.include
 
-        promises.push(@storageManager.loadObject(collectionName, loadOptions))
+          promises.push(@storageManager.loadObject(collectionName, loadOptions))
```

Comment:
FYI: We might want to consider making this procedure more functional. Instead of for-looping and mutating the promises list on every iteration, we should transform the list of batches into an array of promises (or whatever) and then mutate the promises list once.